### PR TITLE
Transport and Persistence initialization setup

### PR DESCRIPTION
**Pull request by ramonsmits**
For SQL Transport and NHibernate persistence the schema's are created and all tables within that schema are truncated.

For ASB it had topology cleanup logic but that is currently not executed due to those operations being scheduled in the Azure back-end and that is takes a long time before those operations are run and there is not a way to detect when those operations are complete.

### CODE REVIEW

File: src/PerformanceTests/Utils/ConfigurationHelper.cs [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
+    public static string FetchSetting(string key)
     {
         string value;
 
         value = Environment.GetEnvironmentVariable(key);
 
         if (!string.IsNullOrWhiteSpace(value))
         {
-            Log.InfoFormat("Setting: {0} = {1} ({2})", key, value, "Environment");
+            Log.Info("Setting: {0} = {1} ({2})", key, value, "Environment");
```
Comment (hmemcpy) [inline src/PerformanceTests/Utils/ConfigurationHelper.cs]: Aren't the `XxxFormat` methods required for the actual string formatting?

Comment (dvdstelt) [inline src/PerformanceTests/Utils/ConfigurationHelper.cs]: If so, it wouldn't compile. This is the NLog version :)

Comment (ramonsmits) [inline src/PerformanceTests/Utils/ConfigurationHelper.cs]: indeed, I moved the class to the Utils project and that does not referenced V5 or V6. Transitioned to NLog which has a slightly different signate.

File: src/PerformanceTests/Persistence.V5.NHibernate.V6/NHibernateProfile.cs [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
-class NHibernateProfile : IProfile
+class NHibernateProfile : IProfile, ISetup
 {
     public void Configure(BusConfiguration cfg)
     {
         cfg
             .UsePersistence<NHibernatePersistence>()
-            .ConnectionString(this.GetConnectionString("NHibernate"))
+            .ConnectionString(ConfigurationHelper.GetConnectionString("NHibernate"))
             //.EnableCachingForSubscriptionStorage(TimeSpan.FromSeconds(5))
```
Comment (dvdstelt) [inline src/PerformanceTests/Persistence.V5.NHibernate.V6/NHibernateProfile.cs]: Can this be removed?

Comment (ramonsmits) [inline src/PerformanceTests/Persistence.V5.NHibernate.V6/NHibernateProfile.cs]: Not part of this PR but I've removed those commented lines.

File: src/PerformanceTests/Persistence.V6.NHibernate.V7/NHibernateProfile.cs [hunk 3]
```diff
@@ -1,14 +1,22 @@
-﻿using NServiceBus;
+﻿using System.Reflection;
+using NServiceBus;
 using NServiceBus.Persistence;
 
-class NHibernateProfile : IProfile
+class NHibernateProfile : IProfile, ISetup
```
Comment (dvdstelt) [inline src/PerformanceTests/Persistence.V6.NHibernate.V7/NHibernateProfile.cs]: We need more interfaces, gotta love interfaces! :)

File: src/PerformanceTests/Persistence.V6.NHibernate.V7/init.sql [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿IF NOT EXISTS (SELECT * FROM sys.schemas WHERE name = 'V6')
+BEGIN
+	EXEC('CREATE SCHEMA V6')
+END
+
+EXEC sp_msforeachtable @command1='TRUNCATE TABLE ?',@whereand='AND Schema_Id = Schema_id(''V6'')'
```
Comment (dvdstelt) [inline src/PerformanceTests/Persistence.V6.NHibernate.V7/init.sql]: Nice trick

File: src/PerformanceTests/Transport.V5.RabbitMQ.V3/RabbitMQProfile.cs [hunk 5]
```diff
@@ -7,6 +7,6 @@ public void Configure(BusConfiguration busConfiguration)
         busConfiguration
             .UseTransport<RabbitMQTransport>()
             //.DisableCallbackReceiver()
```
Comment (dvdstelt) [inline src/PerformanceTests/Transport.V5.RabbitMQ.V3/RabbitMQProfile.cs]: Can this be removed?

File: src/PerformanceTests/Persistence.V6.NHibernate.V7/NHibernateProfile.cs [hunk 6]
```diff
@@ -1,14 +1,22 @@
...
 {
     public void Configure(EndpointConfiguration cfg)
     {
         cfg
             .UsePersistence<NHibernatePersistence>()
-            .ConnectionString(this.GetConnectionString("NHibernate"))
+            .ConnectionString(ConfigurationHelper.GetConnectionString("NHibernate"))
             //.EnableCachingForSubscriptionStorage(TimeSpan.FromSeconds(5))
```
Comment (dvdstelt) [inline src/PerformanceTests/Persistence.V6.NHibernate.V7/NHibernateProfile.cs]: Can this be removed?

File: src/PerformanceTests/Utils/ConfigurationHelper.cs [hunk 7]
```diff
@@ -1,47 +1,47 @@
 ﻿using System;
 using System.Configuration;
-using NServiceBus.Logging;
+using NLog;
 
-public static class ProfileExtensionMethods
+public static class ConfigurationHelper
```
Comment (WilliamBZA) [inline src/PerformanceTests/Utils/ConfigurationHelper.cs]: :+1:

File: src/PerformanceTests/Persistence.V6.NHibernate.V7/init.sql (hunk 4 above)
Comment (ramonsmits) [inline src/PerformanceTests/Persistence.V6.NHibernate.V7/init.sql]: Only tables where this might not work are complex saga aggregates.
